pkg/controller/jobframework/validation.go

```diff
@@ -71,3 +73,14 @@ func ValidateUpdateForParentWorkload(oldJob, newJob GenericJob) field.ErrorList
 	}
 	return allErrs
 }
+
+func ValidateUpdateForOriginalNodeSelectors(oldJob, newJob GenericJob) field.ErrorList {
+	var allErrs field.ErrorList
+	if !oldJob.IsSuspended() && !newJob.IsSuspended() {
```

Comment:
nit: I don't think it matters for business cases, but we could make the job immutable unless it is suspending or unsuspending, it could be expressed easily:
```golang
if oldJob.IsSuspended() != newJob.IsSuspended() {
```
If you go for this, please also adjust the error message accordingly, for example: "this annotation is immutable unless the job is being suspended or unsuspended" (started or stopped)